person.rb

```diff
@@ -23,3 +30,39 @@ def generate_id
     rand(1_000_000)
   end
 end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super() # This is required for linter checks
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  MAX_LENGTH = 10
+
+  def correct_name
+    trimmed_name = @nameable.correct_name[0, MAX_LENGTH]
+    trimmed_name.length == MAX_LENGTH ? trimmed_name : "#{trimmed_name}..."
+  end
+end
```

Comment:
- [ ] Good job creating the decorator feature in your school library 👏 . You have created all the classes in one file , the person.rb file, for the sake of good practices, you should create each `class` in a different file 👍. Kindly create a file for the `BaseDecorator`, `CapitalizeDecorator`, and for the `TrimmerDecorator`. 😄